### ISSUE TITLE
Can the data in markdown syntax be used to create a table view and sort my email notes by date?

### ISSUE THREAD

**Issue by Dricc123** [marph91/thunderbird-joplin-export#26]
It's more of a question than an issue and please mark it as closed when done.
Your add-on is great! I especially like that email metadata gets transfered to the body of the note in a customized way either as text or as table. Now I'm wondering: can this data be used to display notes in a certain way? Let me explain: Obsidian has a Dataview plugin that uses the notes' YAML Frontmatter data to create views of multiples notes, typically a list view or a table view. For example with email notes, I can think of a table view of my email notes with the date of the email in column 1 and the title of the note in column 2. I would also be able to sort the dates chronologically.
Joplin doesn't have such a plug-in at this time. But what if I open my email notes in Obsidian? Will Obsidian take it as YAML Frontmatter metadata and be able to create views of notes?

**Comment by marph91** [comment]
> For example with email notes, I can think of a table view of my email notes with the date of the email in column 1 and the title of the note in column 2.

You could take a look at the "[Note overview](https://github.com/JackGruber/joplin-plugin-note-overview?tab=readme-ov-file#examples)" plugin. You can create a summary note from a set of filtered notes. I will check if it's possible to set the created time from the Thunderbird plugin. Then you could use the "Note overview" plugin to display a list with date and title. You could probably do this already, but the date will be the date of import, not the date when the mail was sent/received.

> But what if I open my email notes in Obsidian? Will Obsidian take it as YAML Frontmatter metadata and be able to create views of notes?

If it's correctly formatted, Obsidian should be able to create the view as you want. Not sure though, since I don't use Obsidian.

**Comment by Dricc123** [comment]
> I will check if it's possible to set the created time from the Thunderbird plugin. Then you could use the "Note overview" plugin to display a list with date and title.

Not sure what you mean. We would need the date of the email, if I'm not mistaken.

But you fully understood what I wanted to achieve and the Note overview plug-in looks good! from the link you sent I understand the syntax would need to start with `<!-- note-overview-plugin` so the plugin knows where to take the data.

So nice of you to check this for me. Thank you.

**Comment by marph91** [comment]
> > I will check if it's possible to set the created time from the Thunderbird plugin. Then you could use the "Note overview" plugin to display a list with date and title.
> 
> Not sure what you mean. We would need the date of the email, if I'm not mistaken.

Sorry, that were implementation details. There are two relevant timestamps for this use case in Joplin's metadata:

- `created_time`: This is the time when the note was created in Joplin.
- `user_created_time`: This is currently unused, but could be set by the Thunderbird plugin.

AFAIK you can use both of the timestamps in the note overview plugin. (Untested) example to show all notes with the `email` tag chronologically:

```
<!-- note-overview-plugin
search: tag:email
fields: created_time, title
sort: created_time DESC
-->
```

> So nice of you to check this for me. Thank you.

You are welcome. I'm using this plugin myself, so it's not much effort :)

**Comment by Dricc123** [comment]
Understood. Very clear.
I'll be back end of next week as I'm off for a couple of days. just in case you wonder why I'm silent.

**Comment by Dricc123** [comment]
> * `user_created_time`: This is currently unused, but could be set by the Thunderbird plugin. (Not yet implemented.)

So you checked and confirm it's feasible?

**Comment by marph91** [comment]
Yes, it's feasible. In fact, I just submitted a new version of the add-on for review at Mozilla. After the review is done, you could download the new version and check if it works for your use case.

**Comment by Dricc123** [comment]
It's working great! My notes are sorted by the Email Date in my note generated by Note Overview plugin. Just what I needed. thank you!

**Comment by Dricc123** [comment]
Oh sorry, I was too fast. I don't understand why but my note-overview note displays the updated time in the first column although I set it up to display the created time (I copy-pasted your above code)...

**Comment by marph91** [comment]
I adapted the example to use `user_created_time`. It should work now.

**Comment by Dricc123** [comment]
Works perfectly with the Overview Plugin now. Thank you very much!

**Comment by Dricc123** [comment]
Hello, slightly off topic: I had an LLM create a python script to convert EML to MD. It goes smoothly for the MD file creation but when I import these files into Joplin, the creation date is not the date of the email, despite telling the LLM that the email date must go as `user_created_time` in the metadata. Can you think of another way to explain him?

**Comment by marph91** [comment]
Hi, this is indeed off topic, since the API (used here) and the Markdown import (used in your script, I guess) are separate things. In general, you can add metadata in [YAML front matter](https://joplinapp.org/help/dev/spec/interop_with_frontmatter/). You might take a look how [jimmy](https://github.com/marph91/jimmy) does it. It supports the [EML to Markdown conversion](https://marph91.github.io/jimmy/formats/default/), too.

For further questions the [Joplin forum](https://discourse.joplinapp.org/) would be better suited.

**Comment by Dricc123** [comment]
Oh, I didn't know Jimmy. It's nice!
Still, my issue is the same with Jimmy.
I'd like it to be like when I use Thunderbird-joplin-export add-on. I can sort my emails by date in Joplin, because the created date shows as the date of the email. I have no idea how you managed to make it do that. What's the trick?

**Comment by marph91** [comment]
Here it's done by using Joplin's data API: https://github.com/marph91/thunderbird-joplin-export/blob/cfa5e9d8d3037016b3b2ef4fc4470dd832c8ddec/src/background.ts#L239

With Jimmy, it should work if you use the `--frontmatter joplin` argument and import to Joplin by "File -> Import -> MD - Markdown + Front Matter"

**Comment by Dricc123** [comment]
Thank you very much. With Jimmy and using `--frontmatter joplin` it works great. You made my day!

With the Jimmy method I noted a little difference compared to Thunderbird-joplin-export in regards to the updated date: with Jimmy it is set to the date of the email whereas with the add-on it is set to the date of the Thunderbird to Joplin operation. It's all fine, right?

**Comment by marph91** [comment]
> With the Jimmy method I noted a little difference compared to Thunderbird-joplin-export in regards to the updated date: with Jimmy it is set to the date of the email whereas with the add-on it is set to the date of the Thunderbird to Joplin operation. It's all fine, right?

Actually, only the created date is set in the Thunderbird addon. I think the updated time should be set, too. In the mean time, you could use the created date.
